gcp/api/server.py

```diff
@@ -710,52 +709,59 @@ def do_query(query: osv_service_v1_pb2.Query,
              context: QueryContext,
              include_details=True) -> tuple[list, str | None]:
   """Do a query."""
+  package_name = ''
+  ecosystem = ''
+  purl_str = ''
+  version = ''
+
   if query.HasField('package'):
     package_name = query.package.name
     ecosystem = query.package.ecosystem
     purl_str = query.package.purl
-  else:
-    package_name = ''
-    ecosystem = ''
-    purl_str = ''
 
-  if ecosystem and not ecosystems.get(ecosystem):
-    context.service_context.abort(grpc.StatusCode.INVALID_ARGUMENT,
-                                  'Invalid ecosystem.')
+  if query.HasField('version'):
+    version = query.version
 
-  purl = None
-  purl_version = None
+  # convert purl to package names
   if purl_str:
     try:
-      purl = PackageURL.from_string(purl_str)
-      purl_version = purl.version
+      purl = purl_helpers.parse_purl(purl_str)
     except ValueError:
```

Comment:
We probably want to let them know the exact reason their PURL is invalid.

Reply:
Discussed offline, we'll keep the current invalid PURL error message for now. Our integration tests rely on error messages, and providing detailed PURL parsing failure reasons doesn't adding too much benefits.